src/dialect/mod.rs

```diff
@@ -248,4 +257,98 @@ mod tests {
     fn parse_dialect(v: &str) -> Box<dyn Dialect> {
         dialect_from_str(v).unwrap()
     }
+
+    #[test]
+    fn parse_with_wrapped_dialect() {
+        /// Wrapper for a dialect. In a real-world example, this wrapper
```

Comment:
👨‍🍳 👌  -- very nice example of creating a test illustrating the use case

Reply:
Thank you!